core/src/proxy/block_engine_stage.rs

```diff
@@ -148,7 +151,13 @@ impl BlockEngineStage {
         while !exit.load(Ordering::Relaxed) {
             // Wait until a valid config is supplied (either initially or by admin rpc)
             // Use if!/else here to avoid extra CONNECTION_BACKOFF wait on successful termination
-            let local_block_engine_config = block_engine_config.lock().unwrap().clone();
+            let local_block_engine_config = {
+                let block_engine_config = block_engine_config.clone();
+                task::spawn_blocking(move || block_engine_config.lock().unwrap().clone())
+                    .await
+                    .map_err(ProxyError::TokioJoinError)
```

Comment:
i dnt think we need this error. if this panicks then some other thread had panicked and poisoned the lock, seems pretty idiomatic to just unwrap here

Reply:
remove map_err and just unwrap

Reply:
there's a panic handler in the client right?

Reply:
agree on generally avoiding unwraps, but in this case a panic would mean some other thread panicked and poisoned this lock. this seems fine to me and idiomatic across rust codebases

Reply:
Also the JoinError only occurs when the tokio task is cancelled, or the sync code panics: https://docs.rs/tokio/latest/tokio/task/struct.JoinError.html